src/main/webapp/app/tasks/task-search/task-search.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { LocalTaskService } from 'app/tasks/local-task.service';
+import { Subscription } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+
+@Component({
+  selector: 'tiny-task-search',
+  templateUrl: './task-search.component.html',
+  styleUrls: ['./task-search.component.scss']
+})
+export class TaskSearchComponent implements OnInit, OnDestroy {
+  searchForm: FormGroup = new FormGroup({
+    text: new FormControl('')
+  });
+
+  subscriptions = new Subscription();
+  @Output() searchEvent = new EventEmitter<string>();
+
+  constructor() { }
+
+  ngOnInit(): void {
+    this.bindEvents();
```

Comment:
no real need to extract this into a seperate method, also _bindEvents_ doesn't tell a whole lot about what is happening here

Reply:
I used to use this way usually in components with big forms, so maybe it should be called bindSaveEvents and generally, there could be a lot of logic like subscribe on each control in the form and each group and its controls and some logic inside these subs...
usually for live save on different controls